nx/lib/nx/defn/evaluator.ex

```diff
@@ -535,36 +535,105 @@ defmodule Nx.Defn.Evaluator do
   end
 
   defp format_node_info(%Expr{id: id, op: op, args: args}, res, inspect_limit) do
-    args =
-      Enum.map(
-        args,
-        &inspect(&1, custom_options: [print_id: true], limit: inspect_limit)
-      )
+    id_str = :erlang.ref_to_list(id) |> List.to_string() |> String.replace(["#Ref<", ">"], "")
 
-    result_str = inspect(res, limit: inspect_limit)
+    inspect_opts =
+      case inspect_limit do
+        nil -> []
+        limit -> [limit: limit]
+      end
+
+    args_code =
+      args
+      |> Enum.map(fn arg ->
+        inspected =
+          arg
+          |> inspect(inspect_opts)
+          |> String.trim()
+
+        "  #{inspect(inspected)}"
+      end)
+      |> Enum.join(",\n")
+
+    # Format result as serialized tensor
+    result_code = "result = #{serialize_tensor(res)}"
+
+    """
+    node_id = "#{id_str}"
+    operation = "#{inspect(op)}"
 
-    import Inspect.Algebra
+    args = [
+    #{args_code}
+    ]
+
+    # Result:
+    #{result_code}
+    """
+  end
+
+  defp serialize_tensor(%Nx.Tensor{data: %Expr{id: id}} = _tensor) when is_reference(id) do
+    # This is an unevaluated expression, not a concrete tensor
+    # Show the Node ID so users can find which file contains this tensor
+    id_str = :erlang.ref_to_list(id) |> List.to_string() |> String.replace(["#Ref<", ">"], "")
```

Comment:
Extract this line to a defp so we can use it when defining the filename on line 636 (but keep the replace with _ there too)